lib/manager/components/LabelEditor.js

```diff
@@ -0,0 +1,231 @@
+ */
+class LabelEditor extends React.Component<Props, State> {
+    state = {
+      validation: {
+        name: true,
+        description: true
+      },
+      newLabel: {...this.props.label},
+      labelIsNew: false
+    }
+
+    componentDidMount = () => {
+      // If we didn't get a label passed in, declare the label to be new
+      if (Object.keys(this.props.label).length === 0) {
+        this.setState({
+          labelIsNew: true
+        })
+      }
+    }
+
+    userIsAdmin = (user: ManagerUserState) => {
+      const {permissions} = user
+      if (!permissions) return false
+      else return permissions.isApplicationAdmin() || permissions.canAdministerAnOrganization()
+    }
+
+  _onFormChange = ({target}: {target: HTMLInputElement}, alwaysValid: boolean) => {
+    const {name, value, type, checked} = target
+
+    const universalValue: string | boolean = type === 'checkbox' ? checked : value
+    // extra check for flow
+    const valid =
+      alwaysValid ||
+      (typeof universalValue === 'string' && universalValue.length > 0)
+
+    this.setState(
+      update(this.state, {
+        newLabel: { $merge: {[name]: universalValue} },
+        validation: { [name]: { $set: universalValue && valid } }
+      })
+    )
+  }
+
+  /* Compares the dirty label from the old one to determine
+  which keys changed
+  */
+  _getChanges = () => {
+    const newLabel = this.state.newLabel
+    const oldLabel = this.props.label
+
+    const changes: any = {}
+    Object.keys(newLabel).forEach(k => {
+      if (newLabel[k] !== oldLabel[k]) {
+        changes[k] = newLabel[k]
+      }
+    })
+    return changes
+  }
+
+  _settingsAreUnedited = () => Object.keys(this._getChanges()).length === 0
+
+  _formIsValid = () => {
+    const {validation} = this.state
+    return Object.keys(validation).every(k => validation[k])
+  }
+
+  _onSaveSettings = () => {
+    const {labelIsNew, newLabel: label} = this.state
+    const {createLabel, updateLabel} = this.props
+
+    if (labelIsNew) {
+      // Insert project ID
+      label.projectId = this.props.projectId
+      createLabel(label)
+    } else {
+      updateLabel(label)
+    }
+
+    // This passed method will close the editor
+    this.props.onDone && this.props.onDone()
+  }
+
+  render () {
+    const { newLabel, validation } = this.state
+    const { user } = this.props
+
+    return (
+      <Form>
+        <Panel>
+          <ListGroup fill>
+            <ListGroupItem>
+              <FormGroup validationState={validationState(validation.name)}>
+                <ControlLabel>Name</ControlLabel>
+                <FormControl
+                  value={newLabel.name}
+                  name={'name'}
+                  onChange={this._onFormChange}
+                />
+                <FormControl.Feedback />
+                <HelpBlock>Required.</HelpBlock>
+              </FormGroup>
+              <FormGroup>
+                <ControlLabel>Description</ControlLabel>
```

Comment:
When editing an existing label, changing fields other than Description should also enable the Save button.

Reply:
This was a good catch! I've refactored the checkbox validation to resolve this